components/atoms/details2/index.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * Components - Details2
+ */
+
+import { useEffect, useRef, useState } from 'react'
+
+// UI
+import { DetailsPropTypes, DetailsDefaultProps } from './props'
+import {
+  DetailsSubscriber,
+  Icon,
+  MessageNames,
+  SIZE,
+  Text,
+  useComponentCommunication
+} from 'components'
+
+// Style
+import styled from 'styled-components'
+
+export const Details2 = ({
+  animationDuration,
+  children,
+  content,
+  context,
+  endActionComponent,
+  icon,
+  open,
+  startActionComponent,
+  title,
+  toolbar,
+  uniqueId,
+  unmountContentOnClose,
+  ...props
+}) => {
+  const [IsOpen, setIsOpen] = useState(open)
+  const [ChildrenMounted, setChildrenMounted] = useState(unmountContentOnClose ? open : true)
+  const [ContentHeight, setContentHeight] = useState(0)
+
+  const contentRef = useRef(null)
+
+  useEffect(() => {
+    if (unmountContentOnClose) {
+      IsOpen
+        ? setChildrenMounted(() => true)
+        : setTimeout(() => {
+            window &&
+              window.requestAnimationFrame(() => {
+                setChildrenMounted(() => false)
+              })
+          }, animationDuration ?? 300)
+    }
+    setTimeout(
+      () => {
+        window &&
+          window.requestAnimationFrame(() => {
+            contentRef.current && setContentHeight(() => contentRef.current.offsetHeight)
+          })
+      },
+      !IsOpen ? animationDuration ?? 300 : 0
+    )
+
+    return () => {}
+  }, [contentRef.current, IsOpen])
+
+  const handleEventRecieve = e => {
+    setIsOpen(e)
+  }
+
+  useComponentCommunication({
+    id: uniqueId,
+    messageName: MessageNames.DetailsComponent.SET_OPEN,
+    onRecieve: e => handleEventRecieve(e),
+    subscriber: DetailsSubscriber
+  })
+
+  const handleOpenClose = () => {
+    setIsOpen(IsOpen => !IsOpen)
+  }
+
+  return (
+    <Wrapper context={context} {...props}>
+      <Header>
+        <OpenIconAndTitleWrapper>
+          {(content || children) && (
+            <OpenCloseWrapper
+              open={IsOpen}
+              onClick={handleOpenClose}
+              animationDuration={animationDuration}
+            >
+              {icon ? <Icon icon={icon} prefix='fas' /> : <CaretRight />}{' '}
```

Comment:
What if the icon had prefix? Isn't it better to have a render prop also for icon instead of just passing the icon name? I mean like what you did for `startActionComponent`, something like `iconComponent`.